packages/xrpl/test/integration/requests/getAggregatePrice.test.ts

```diff
@@ -0,0 +1,79 @@
+import { stringToHex } from '@xrplf/isomorphic/dist/utils'
+import { assert } from 'chai'
+
+import { OracleSet } from '../../../src'
+import { getUnixTimestamp } from '../../../src/utils/timeConversion'
+import serverUrl from '../serverUrl'
+import {
+  setupClient,
+  teardownClient,
+  type XrplIntegrationTestContext,
+} from '../setup'
+import { testTransaction } from '../utils'
+
+// how long before each test case times out
+const TIMEOUT = 20000
+
+describe('get_aggregate_price', function () {
+  let testContext: XrplIntegrationTestContext
+
+  beforeEach(async () => {
+    testContext = await setupClient(serverUrl)
+  })
+  afterEach(async () => teardownClient(testContext))
+
+  it(
+    'base',
+    async () => {
+      const tx: OracleSet = {
+        TransactionType: 'OracleSet',
+        Account: testContext.wallet.classicAddress,
+        OracleDocumentID: 1234,
+        LastUpdateTime: getUnixTimestamp(),
+        PriceDataSeries: [
+          {
```

Comment:
Just a nice-to-have but we could consider adding more `PriceData` to the test series so the aggregation is clearer :)

Reply:
I don't think it matters too much since we're not testing the RPC here, just the library.

Reply:
I agree with @mvadari